File: src/com/kingmang/lazurite/parser/AST/Statements/ExprStatement.java
package com.kingmang.lazurite.parser.AST.Statements;

import com.kingmang.lazurite.parser.AST.Expressions.Expression;
import com.kingmang.lazurite.parser.AST.InterruptableNode;
import com.kingmang.lazurite.patterns.visitor.ResultVisitor;
import com.kingmang.lazurite.patterns.visitor.Visitor;
import com.kingmang.lazurite.runtime.LzrValue;
import lombok.AllArgsConstructor;

@AllArgsConstructor
public final class ExprStatement extends InterruptableNode implements Expression, Statement {
    
    public final Expression expr;

    @Override
    public void execute() {
        super.interruptionCheck();
        expr.eval();
    }
    
    @Override
    public LzrValue eval() {
        return expr.eval();
    }

    @Override
    public void accept(Visitor visitor) {
        visitor.visit(this);
    }

    @Override
    public <R, T> R accept(ResultVisitor<R, T> visitor, T t) {
        return visitor.visit(this, t);
    }

    @Override
    public String toString() {
        return expr.toString();
    }
}

File: src/main/java/com/kingmang/lazurite/parser/ast/statements/ExprStatement.kt
package com.kingmang.lazurite.parser.ast.statements

import com.kingmang.lazurite.parser.ast.InterruptableNode
import com.kingmang.lazurite.parser.ast.expressions.Expression
import com.kingmang.lazurite.patterns.visitor.ResultVisitor
import com.kingmang.lazurite.patterns.visitor.Visitor
import com.kingmang.lazurite.runtime.values.LzrValue

data class ExprStatement(val expr: Expression) : InterruptableNode(), Expression, Statement {
    override fun execute() {
        super.interruptionCheck()
        this.expr.eval()
    }

    override fun eval(): LzrValue =
        this.expr.eval()

    override fun accept(visitor: Visitor) =
        visitor.visit(this)

    override fun <R, T> accept(visitor: ResultVisitor<R, T>, input: T): R? =
        visitor.visit(this, input)

    override fun toString(): String =
        this.expr.toString()
}
